elasticgraph-graphql/spec/unit/elastic_graph/graphql/datastore_query/filtering_spec.rb

```diff
@@ -364,6 +364,116 @@ class GraphQL
       end
 
       describe "`all_of` operator" do
+        context "top-level usage on distinct fields" do
+          it "behaves the same as specifying multiple fields in the same object" do
+            query = new_query(filter: {"all_of" => [
+              {"color" => {"equal_to_any_of" => ["RED"]}},
+              {"size" => {"gt" => 10}}
+            ]})
+
+            expect(datastore_body_of(query)).to query_datastore_with(bool: {
+              filter: [
+                {
+                  bool: {
+                    filter: [
+                      {terms: {"color" => ["RED"]}}
+                    ]
+                  }
+                },
+                {
+                  bool: {
+                    filter: [
+                      {range: {"size" => {gt: 10}}}
+                    ]
+                  }
+                }
+              ]
+            })
+          end
+        end
+
+        context "top-level usage on the same key" do
+          it "allows ANDing two subfilters on the same field (otherwise impossible in a single object)" do
+            query = new_query(filter: {
+              "all_of" => [
+                {"color" => {"equal_to_any_of" => ["RED"]}},
+                {"color" => {"not" => {"equal_to_any_of" => ["GREEN"]}}}
+              ]
+            })
```

Comment:
I believe this is possible w/o `all_of`:

```ruby
filter: {
  "color" => {
    "equal_to_any_of" => ["RED"],
    "not" => {"equal_to_any_of" => ["GREEN"]}
  }
}
```

...so it's a bit confusing to read it's otherwise impossible in your test description.

Maybe just omit that part of the test description?  While users don't need to use `all_of` to achieve this filtering behavior, they can use it, and you're testing what happens when they do.